File: cpp-leetcode/easy/0014-最长公共前缀.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string& s0 = strs[0];
        for (int j = 0; j < s0.size(); j++) {
            for (string& s : strs) {
                if (j == s.size() || s[j] != s0[j]) { 
                    return s0.substr(0, j);
                }
            }
        }
        return s0;
    }
};

File: python-leetcode/0014-最长公共前缀.py

class Solution:
    def longestCommonPrefix(self, strs):
        s0 = strs[0]
        for j in range(len(s0)):
            for s in strs:
                if j == len(s) or s[j] != s0[j]:
                    return s0[:j]
        return s0
